lettuce/cuda_native/_template.py

```diff
@@ -212,27 +253,18 @@ def invoke(simulation):
   }};
 
 #if {support_no_collision_mask}
-  const index_t node_index = node_coord(index[0], index[1], index[2]);
+  const index_t node_index = node();
 #endif
 
-#if {support_no_streaming_mask}
   index_t dist_index[q];
-#pragma unroll
-  for (index_t i = 0; i < q; ++i) {{
-    dist_index[i] = dist_coord(i, index[0], index[1], index[2]);
-  }}
-  scalar_t f_reg[q];
 #pragma unroll
   for (index_t i = 0; i < q; ++i)
-    f_reg[i] = f[dist_index[i]];
-#else
+    dist_index[i] = distribution(i);
+
   scalar_t f_reg[q];
+#pragma unroll
   for (index_t i = 0; i < q; ++i)
-    f_reg[i] = f[dist_coord(i, index[0], index[1], index[2])];
-#endif
-
-  constexpr index_t e[q][d] = {e};
-  constexpr scalar_t w[q] = {w};
+    read(i);
```

Comment:
key change: simplified template body by shifting the streaming logic into a macro definition